### PR TITLE
mg: new recipe

### PR DESCRIPTION
**Pull request by hoanga**
the following recipe adds the mg editor

sample post-install below:

```
> uname -a
Haiku shredder 1 hrev56085 Apr 29 2022 07:35:01 x86_64 x86_64 Haiku

> man mg | head
MG(1)                       General Commands Manual                      MG(1)

NAME
     mg - emacs-like text editor

SYNOPSIS
     mg [-nR] [-b file] [-f mode] [-u file] [+number] [file ...]

DESCRIPTION
     mg is intended to be a small, fast, and portable editor for people who

> which mg
/bin/mg
```

### CODE REVIEW

File: app-editors/mg/mg-7.0.recipe [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+	cmd:mg$secondaryArchSuffix = $portVersion
+	"
+REQUIRES="
+	haiku$secondaryArchSuffix
+	lib:libncursesw$secondaryArchSuffix
+	"
+
+BUILD_REQUIRES="
+	haiku${secondaryArchSuffix}_devel
+        devel:libncursesw$secondaryArchSuffix
```
Comment (Begasus) [inline app-editors/mg/mg-7.0.recipe:30]: 1 tab here, no spaces

File: app-editors/mg/mg-7.0.recipe [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+	cmd:make
```
Comment (Begasus) [inline app-editors/mg/mg-7.0.recipe:31]: Don't think make is required for the devel package (already in BUILD_PREREQUIRES)
```diff
- 	cmd:make
```

File: app-editors/mg/mg-7.0.recipe [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+	"
+BUILD_PREREQUIRES="
+	cmd:gcc$secondaryArchSuffix
+	cmd:make
+	"
+
+BUILD()
+{
+	runConfigure ./configure --prefix="$prefix"
+	make
```
Comment (Begasus) [inline app-editors/mg/mg-7.0.recipe:41]: No need for prefix, runConfigure takes care of that:
```diff
- 	runConfigure ./configure --prefix="$prefix"
-  	make
+ 	runConfigure ./configure
+ 	make $jobArgs
```

Comment (Begasus) [reply app-editors/mg/mg-7.0.recipe:40]: Didn't check a build yet, but I guess you could use $jobArgs here :)

Comment (Begasus) [reply app-editors/mg/mg-7.0.recipe]: Only prefix left, will do a build here to check :)

File: app-editors/mg/mg-7.0.recipe [hunk 4]
```diff
@@ -0,0 +1,46 @@
+SUMMARY="Portable Public Domain Micro Emacs"
+DESCRIPTION="Mg is intended to be a small, fast, and portable editor for \
+people who can't (or don't want to) run emacs for one reason or another, \
+or are not familiar with the vi editor."
+HOMEPAGE="https://github.com/ibara/mg"
+COPYRIGHT="2000"
```
Comment (Begasus) [inline app-editors/mg/mg-7.0.recipe:6]: Maybe:
```diff
- COPYRIGHT="2000"
+ COPYRIGHT="2019-2021 Brian Callahan"
```

File: app-editors/mg/mg-7.0.recipe [hunk 5]
```diff
@@ -0,0 +1,46 @@
...
+
+ARCHITECTURES="all !x86_gcc2"
+SECONDARY_ARCHITECTURES="x86"
+
+libVersion="$portVersion"
+libVersionCompat="$libVersion compat >= 1"
+
+PROVIDES="
+	mg$secondaryArchSuffix = $portVersion
+	cmd:mg$secondaryArchSuffix = $portVersion
```
Comment (Begasus) [inline app-editors/mg/mg-7.0.recipe:21]: cmd is installed in $prefix/bin, so:
```diff
- 	cmd:mg$secondaryArchSuffix = $portVersion
+ 	cmd:mg = $portVersion
```

File: app-editors/mg/mg-7.0.recipe [hunk 6]
```diff
@@ -0,0 +1,46 @@
...
+REVISION="1"
+SOURCE_URI="https://github.com/ibara/mg/releases/download/mg-$portVersion/mg-$portVersion.tar.gz"
+CHECKSUM_SHA256="650dbdf9c9a72ec1922486ce07112d6181fc88a30770913d71d5c99c57fb2ac5"
+PATCHES="mg-$portVersion.patchset"
+
+ARCHITECTURES="all !x86_gcc2"
+SECONDARY_ARCHITECTURES="x86"
+
+libVersion="$portVersion"
+libVersionCompat="$libVersion compat >= 1"
```
Comment (Begasus) [inline app-editors/mg/mg-7.0.recipe:17]: Probably a leftover?
```diff
-libVersion="$portVersion"
-libVersionCompat="$libVersion compat >= 1"
```

**Comment by Begasus** [comment]
After looking at some recipes through repology, I see a lot of them using https://github.com/hboetes/mg for the sources
Even Gentoo does: https://packages.gentoo.org/packages/app-editors/mg

**Comment by hoanga** [comment]
thanks for the quick feedback and i had a brief look at the hboetes version of mg and it seems that it uses a different approach for the port and would require another library (libbsd) to be available. the tagging scheme for each of the other cited mg ports are also different (hboetes uses timestamp, troglobit follows semver but at a different version).

for the time being, i have pushed an update that addresses suggestions in case this flavor of the mg port is chosen.

**Comment by Begasus** [comment]
Thanks!

**Pull request merged**